### PR TITLE
Prefer returning arrays instead of hashes

### PR DESCRIPTION
**Pull request by marcandre**
This is what I think it should look like. How does it look to you?

Fixes [#17]

### CODE REVIEW

**Comment by marcandre** [comment]
Rebased and tweaked after recent merges.

File: lib/tlaw/namespace.rb [hunk 1]
```diff
@@ -48,25 +49,50 @@ def base_url=(url)
...
+          fail ArgumentError, "#{name} is not an #{restrict_to}" unless ep < (restrict_to || Object)
+        end
+      end
+
+      # Lists all current namespace's nested namespaces.
+      #
+      # @return [Namespace, ...]
       def namespaces
-        children.select { |_k, v| v < Namespace }
+        children.grep(Namespace.singleton_class)
```
[the remark this answers is not in the document]
Comment (marcandre) [reply lib/tlaw/namespace.rb]: I was actually fearing your reaction on that one! It's pretty hacky and definitely not beginner friendly, but for advanced Rubyists, it should be understandable.

File: lib/tlaw/namespace.rb [hunk 2]
```diff
@@ -48,25 +49,50 @@ def base_url=(url)
...
           child.traverse(restrict_to, &block) if child.respond_to?(:traverse)
         end
         self
       end
 
-      # Lists all current namespace's nested namespaces as a hash.
+      # Returns the namespace's child of the requested name.
       #
-      # @return [Hash{Symbol => Namespace}]
+      # @return [Endpoint, ...]
```
[the remark this answers is not in the document]
Comment (marcandre) [reply lib/tlaw/namespace.rb]: Oups. Fixed.

File: lib/tlaw/namespace.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (zverok) [reply lib/tlaw/namespace.rb]: I definitely like it much more than `select` + `<`, wasn't aware of the trick myself but it makes me happy :) 
As for "beginner-friendliness"—I believe all ways of DRYing/making code shorter are great, as long as they are readable and logical (And I am pretty grumpy when people say "it is not readable" about absolutely clean code in fact meaning "I never used this method" or "Junior developer would need to google this method call sequence")

**Comment by zverok** [comment]
Thanks!

**Pull request merged**